mathics/builtin/strings.py

```diff
@@ -623,6 +623,70 @@ class HexidecimalCharacter(Builtin):
     """
 
 
+class DigitQ(Builtin):
+    """
+    <dl>
+    <dt>'DigitQ[$string$]'
+        yields 'True' if all the characters in the $string$ are digits, and yields 'False' otherwise. 
+    </dl>
+
+    >> DigitQ["9"]
+     = True
+
+    >> DigitQ["a"]
+     = False
+
+    >> DigitQ["01001101011000010111010001101000011010010110001101110011"]
+     = True
+
+    >> DigitQ["-123456789"]
+     = False
+
+    #> DigitQ["."]
+     = False
+
+    #> DigitQ[1==2]
+     = False
+
+    #> DigitQ[a=1]
+     = False
+    """
+
+    rules = {
+        'DigitQ[string_]': (
+            'If[StringQ[string], StringMatchQ[string, DigitCharacter..], False, False]'),
```

Comment:
I think it should be `DigitCharacter...` for compatibility.  Could you also add a test for this `DigitQ[""]` (`True`).